### PR TITLE
created Bookmark selection

### LINKED ISSUE #163: DRAFT: Bookmark selection (see screengrab)
When user clicks bookmark icon to save item, box with podcast titles and checkbox appears for user to select which podcast(s) LONGER the item is sent t
Screenshot 2022-05-02 at 06 48 02
o

### PR DESCRIPTION
**Pull request by bhavinsen**
Fixed #163

### CODE REVIEW

File: src/pages/Home.vue [hunk 1]
```diff
@@ -218,8 +218,23 @@ const events = {
...
     itemStore.updateItem(item, props.podcastId, docname)
   },
-  onClickShare(item: Item, destination: string) {
-    shareStore.sendItem(item, destination, props.podcastId)
+  onClickShare(item: Item, destination: string = "", podcastNameToShare: [],getDeletePodcastId: string = "") {
+    if(destination !== ""){
+      shareStore.sendItem(item, destination, props.podcastId)
+    }
+    if(getDeletePodcastId !== ""){
+      shareStore.deleteItem(item, getDeletePodcastId, props.podcastId)
```
Comment (chrisns) [inline src/pages/Home.vue:226]: shouldn't be able to delete from someone elses inbox

**Review by chrisns — CHANGES_REQUESTED** [review]

**Review by chrisns — APPROVED** [review]

**Review by chrisns — CHANGES_REQUESTED** [review]

File: src/pages/Home.vue [hunk 2]
```diff
@@ -245,8 +245,11 @@ const events = {
   onClickUpdate(item: Item) {
     itemStore.updateItem(item, props.podcastId, docname)
   },
-  onClickShare(item: Item, destination: string) {
+  onClickShare(item: Item, destination: any) {
     shareStore.sendItem(item, destination, props.podcastId)
+    if(destination.includes(props.podcastId)){
+      shareStore.connect(props.podcastId)
```
Comment (chrisns) [inline src/pages/Home.vue:251]: Shouldn't need if already subscribed

File: src/store/itemShare.ts [hunk 3]
```diff
@@ -32,24 +33,22 @@ export const useShareStore = defineStore("share", {
...
+          })
+          // eslint-disable-next-line @typescript-eslint/no-explicit-any
+        } catch (e: any) {
+          // create doc if it doesn't exist
+          if (e.code === "not-found" && e.name === "FirebaseError")
+            await setDoc(docRef, { items: arrayUnion(item.text) })
+          else throw e
+        }
+      })
+      this.connect(from)
```
Comment (chrisns) [inline src/store/itemShare.ts:51]: ?

**Review by chrisns — CHANGES_REQUESTED** [review]

File: src/store/itemShare.ts (hunk 3 above)
Comment (bhavinsen) [reply src/store/itemShare.ts:51]: After switching podcasts in the podcast menu then all connections are mixed so after sharing we need to call the current connection So I have called the "connect" function hear we have removed this function and then got an inbox reordering issue please refer to this issue in my attachment.

https://user-images.githubusercontent.com/13918292/170646958-fcfedac9-e17e-4898-9d28-ec0214fc55cd.mp4

File: src/pages/Home.vue (hunk 2 above)
Comment (bhavinsen) [reply src/pages/Home.vue]: I have removed this code.

**Comment by mrjamieeast** [comment]
This works for me

File: src/store/itemShare.ts (hunk 3 above)
Comment (chrisns) [reply src/store/itemShare.ts:51]: I see, sounds like a work around to a bug, this shouldn't really be needed

Comment (chrisns) [inline src/store/itemShare.ts:51]:
```suggestion
     // TODO: needed due to the issue described https://github.com/DaftDoris/newsdesk/pull/230/files#r883330694 it shouldn't be necessary to re-establish the subscription every time
      this.connect(from)
```

**Review by chrisns — CHANGES_REQUESTED** [review]

**Comment by chrisns** [comment]
@mrjamieeast Please test

**Pull request merged**